internal/stolonctl/pkg/cluster/upgrade.go

```diff
@@ -53,12 +63,65 @@ func Upgrade(ctx context.Context, config Config) error {
 		Name:      defaults.CRDName,
 		Namespace: config.Namespace,
 	}
-	res, err := crdclient.Create(ctx, objMeta)
-	if err != nil && !trace.IsAlreadyExists(err) {
+	res, err := crdclient.CreateOrRead(objMeta)
+	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	// temporary
 	log.Info(res)
+	res, err = crdclient.MarkStepCompleted(res, crd.StolonUpgradeStepInit)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Backup Postgres
+	res, err = crdclient.MarkStepStarted(res, crd.StolonUpgradeStepBackupPostgres)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if err = backupPostgres(config); err != nil {
+		return trace.Wrap(err)
+	}
+	res, err = crdclient.MarkStepCompleted(res, crd.StolonUpgradeStepBackupPostgres)
```

Comment:
`res` is redundant as not used anymore.
I also wonder what happens when you restart the upgrade command - does it rewrite the steps?

Reply:
Ooops, yes, it will rewrite the steps. 
Updated `res` is used in steps below. Or, better to reread from CRD every time?